ClientLib/src/main/java/com/o3dr/services/android/lib/coordinate/LatLongAlt.java

```diff
@@ -53,7 +76,9 @@ public boolean equals(Object o) {
 
         LatLongAlt that = (LatLongAlt) o;
 
-        if (Double.compare(that.mAltitude, mAltitude) != 0) return false;
+        if ((Double.compare(that.mAltitude, mAltitude) != 0)
+            && (that.mFrame.asInt() != mFrame.asInt()) )
```

Comment:
In addition, the `equals()`, `hashCode()` and `toString()` methods were generated automatically via Android Studio: `Code` => `Generate`.